sys-fs/zfs-auto-snapshot/zfs-auto-snapshot-1.2.4.ebuild

```diff
@@ -0,0 +1,61 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/zfsonlinux/${PN}.git"
+else
+	SRC_URI="https://github.com/zfsonlinux/${PN}/archive/upstream/${PV}.tar.gz -> ${P}.tar.gz"
```

Comment:
Could you please use the same distfile name as the directory inside it? i.e. `${PN}-upstream-${PV}`. You can also put that string in common variable `MY_P` for reuse.